pkg/controller/controller_instance_test.go

```diff
@@ -1632,6 +1766,112 @@ func TestPollServiceInstanceSuccessDeprovisioningWithOperationWithFinalizer(t *t
+
+	sharedInformers.ServiceBrokers().Informer().GetStore().Add(getTestServiceBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := getTestServiceInstance()
+	startTime := metav1.NewTime(time.Now().Add(-7 * 24 * time.Hour))
+	instance.Status.OperationStartTime = &startTime
+
+	complete, err := testController.reconcileServiceInstance(instance)
+	if err == nil {
+		t.Fatalf("This should fail, but did not")
+	}
+	if !complete {
+		t.Fatal("Instance reconciliation should complete")
+	}
+
+	brokerActions := fakeServiceBrokerClient.Actions()
+	assertNumberOfServiceBrokerActions(t, brokerActions, 0)
+
+	instanceKey := testNamespace + "/" + testServiceInstanceName
+
+	// Since synchronous operation, must not make it into the polling queue.
+	if testController.pollingQueue.NumRequeues(instanceKey) != 0 {
+		t.Fatalf("Expected polling queue to not have any record of test instance")
+	}
+
+	actions := fakeCatalogClient.Actions()
+	assertNumberOfActions(t, actions, 1)
+
+	updatedServiceInstance := assertUpdateStatus(t, actions[0], instance)
+	assertServiceInstanceCondition(t, updatedServiceInstance, v1alpha1.ServiceInstanceConditionFailed, v1alpha1.ConditionTrue, errorReconciliationRetryTimeoutReason)
+
+	assertServiceInstanceReconciledGeneration(t, updatedServiceInstance, 1)
+	assertServiceInstanceCurrentOperationTimeSet(t, updatedServiceInstance, false)
+
+	kubeActions := fakeKubeClient.Actions()
+	assertNumberOfActions(t, kubeActions, 0)
+
+	events := getRecordedEvents(testController)
+	assertNumEvents(t, events, 1)
+
+	expectedEvent := api.EventTypeWarning + " " + errorReconciliationRetryTimeoutReason
+	if e, a := expectedEvent, events[0]; !strings.Contains(a, e) {
+		t.Fatalf("Received unexpected event:\n  expected to contain: %v\n  got: %v", e, a)
+	}
+}
+
+// TestPollServiceInstanceRetryTimeout tests that reconciliation ends when
+// the retry timeout is reached
+func TestPollServiceInstanceRetryTimeout(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeServiceBrokerClient, testController, sharedInformers := newTestController(t, noFakeActions())
+
+	addGetNamespaceReaction(fakeKubeClient)
+
+	sharedInformers.ServiceBrokers().Informer().GetStore().Add(getTestServiceBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := getTestServiceInstanceAsyncProvisioning("")
+	startTime := metav1.NewTime(time.Now().Add(-7 * 24 * time.Hour))
+	instance.Status.OperationStartTime = &startTime
+
+	complete, err := testController.pollServiceInstanceInternal(instance)
+	if err == nil {
+		t.Fatalf("This should fail, but did not")
+	}
+	if !complete {
+		t.Fatal("Instance reconciliation should complete")
+	}
+
+	brokerActions := fakeServiceBrokerClient.Actions()
+	assertNumberOfServiceBrokerActions(t, brokerActions, 0)
+
+	instanceKey := testNamespace + "/" + testServiceInstanceName
+
+	// Since synchronous operation, must not make it into the polling queue.
```

Comment:
This comment is untrue for this test. The polling queue should not have a record of the instance because it has been removed due to being finished.